### ISSUE TITLE
Turtle writer does not respect namespaces in IRIs

### ISSUE THREAD

**Issue by reckart** [eclipse-rdf4j/rdf4j#4921]
### Current Behavior

The turtle writer does not consider the namespace provided by an IRI. Instead, it turns the IRI into a string and then tries to derive a namespace from that string and then looks up that namespace in the namespace table.

### Expected Behavior

If the IRI defines a namespace, that should first be used to look up the prefix in the namespace table. Interpreting the IRI as a string should only be a fallback.

### Steps To Reproduce

```
	@Test
	public void testIriNamespace() throws Exception {
	    Model model = new DynamicModelFactory().createEmptyModel();
	    String prefix = "foo-bar";
        String ns = "foo:this.is.my.bar.";
        model.setNamespace(prefix, ns);
	    model.add(vf.createIRI(ns, "lala"), vf.createIRI(ns, "lulu"), vf.createIRI(ns, "lolo"));

        StringWriter stringWriter = new StringWriter();
	    Rio.write(model, stringWriter, RDFFormat.TURTLE);
	    
	    assertThat(stringWriter.toString()).contains("foo-bar:lala foo-bar:lulu foo-bar:lolo .");
	}
```

### Version

4.3.9

### Are you interested in contributing a solution yourself?

Yes

**Comment by reckart** [comment]
PR provided: https://github.com/eclipse-rdf4j/rdf4j/pull/4922

**Comment by hmottestad** [comment]
Not seen this before. Can you point to where this is defined in the turtle specification?

**Comment by reckart** [comment]
I have just ported code from Jena to RDF4J. In the Jena version, all IRIs I used were properly abbreviated when a namespace prefix was defined. Not so in RDF4J. I tracked this down to two issues:

1) SimpleIRI does not retain namespace and localname separately (I could work around this using a custom subclass of AbstractIRI)
2) TurtleWriter did not consider the explicit namespace in the IRI

Have a look at the provided test case.

As far as I can see during a quick scan of the Turtle specs, they deal mostly with parsing the turtle format. However, what I am trying to fix here is the serialization. The current serialization done by RDF4J provides valid Turtle, but it is unnecessarily verbose. My change makes more effective use of the namespace prefix declarations, leading to a serialized form that is more compact and easier to read.
